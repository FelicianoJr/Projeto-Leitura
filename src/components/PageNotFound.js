import React from "react";

const PageNotFounds = () => {
  return (
    <div className="jumbotron mt-2 mx-2">
      <h1 className="display-4">Erro 404</h1>
      <p className="lead">Página não encontrada!</p>{" "}
    </div>
  );
};

export default PageNotFounds;
