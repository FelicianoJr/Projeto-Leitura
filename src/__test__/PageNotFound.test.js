import React from "react";
import { shallow } from "enzyme";
import PageNotFound from "../components/PageNotFound";

const setup = data => {
  const component = shallow(<PageNotFound />);

  return {
    component: component,
    display: component.find(".display-4"),
    lead: component.find(".lead")
  };
};

describe("<PageNotFound />", () => {
  it("should render alert", () => {
    const { display, lead } = setup();
    expect(display.text()).toEqual("Oops, :(");
    expect(lead.text()).toEqual("Postagem nÃ£o encontrada!");
  });

  it("snapshot  a PageNotFound", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
