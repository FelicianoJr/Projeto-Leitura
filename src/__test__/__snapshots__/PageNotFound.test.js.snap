// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> snapshot  a PageNotFound 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(NavBarContainer) />,
        <div
          className="jumbotron"
>
          <h3
                    className="display-4"
          >
                    Oops, :(
          </h3>
          <p
                    className="lead"
          >
                    Postagem não encontrada!
          </p>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="display-4"
>
              Oops, :(
</h3>,
            <p
              className="lead"
>
              Postagem não encontrada!
</p>,
          ],
          "className": "jumbotron",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oops, :(",
              "className": "display-4",
            },
            "ref": null,
            "rendered": "Oops, :(",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Postagem não encontrada!",
              "className": "lead",
            },
            "ref": null,
            "rendered": "Postagem não encontrada!",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(NavBarContainer) />,
          <div
            className="jumbotron"
>
            <h3
                        className="display-4"
            >
                        Oops, :(
            </h3>
            <p
                        className="lead"
            >
                        Postagem não encontrada!
            </p>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="display-4"
>
                Oops, :(
</h3>,
              <p
                className="lead"
>
                Postagem não encontrada!
</p>,
            ],
            "className": "jumbotron",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Oops, :(",
                "className": "display-4",
              },
              "ref": null,
              "rendered": "Oops, :(",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Postagem não encontrada!",
                "className": "lead",
              },
              "ref": null,
              "rendered": "Postagem não encontrada!",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
